server/chess.py

```diff
@@ -100,16 +100,17 @@ def hypothetical_check(
                 typing.Tuple[models.Piece, int, int], ...
             ]) -> bool:
         """Check if a series of moves would put a side in check."""
-        if self.hypothetical_moves is None:
+        if moves is None:
             raise RuntimeError('Checkmate detection recursion detected.')
         self.hypothetical_moves = moves    # self.get_piece will observe this
+        """***Note that the hypothetical moves are not yet being used***"""
         king = models.Piece.get(
             models.Piece.side == side,
             models.Piece.piece_type == models.PieceType.KING,
             models.Piece.game == self.game
         )
         enemies = models.Piece.select().where(
-            models.Piece.side == side,
+            models.Piece.side == side.invert,
```

Comment:
```suggestion
            models.Piece.side == ~side,
```
`__invert__` is a "magic" method. It is not meant to be called directly, instead it is an implementation of this unary operator.

Reply:
Yes, I actually realised this later on (like 5am), and thought I'd leave it to you to explain, thanks, makes sense.